src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/CollectDataForReadOrientationFilter.java

```diff
@@ -93,84 +117,46 @@ public void apply(final AlignmentContext alignmentContext, final ReferenceContex
-        // so we must do so manually with a for loop
-        // Median in Apache commons would be an alternative approach, but it takes in a double[], not int[],
-        // so converting to double[] will be just as expensitve as iterating
-        for (final int mq : pileup.getMappingQuals()) {
-            mappingQualities.add(mq);
-        }
-
-        final double medianMQ = MathUtils.median(mappingQualities);
-
-        if (medianMQ < MINIMUM_MEDIAN_MQ) {
-            return;
-        }
-
+        final ReadPileup pileup = alignmentContext.getBasePileup().makeFilteredPileup(pe -> pe.getQual() > MINIMUM_BASE_QUALITY);
         final int[] baseCounts = pileup.getBaseCounts();
         final int depth = (int) MathUtils.sum(baseCounts);
 
-        if (depth == 0) {
-            if (pileup.size() != 0) {
-                logger.info(String.format("Encountered depth = 0 at %s. Pileup size was %d", referenceContext.getInterval(), pileup.size()));
-            }
+        if (! isPileupGood(pileup, referenceContext)){
             return;
         }
 
-        final Nucleotide refBase = Nucleotide.valueOf(refContext.getBytes()[1]);
-
-        // Make a copy of base counts and update the counts of ref to -1. Now the argmax of the array gives us
+        // Make a copy of base counts and update the counts of ref to -1. Now the maxElementIndex of the array gives us
         // the alt base.
+        final Nucleotide refBase = Nucleotide.valueOf(refContext.getBytes()[MIDDLE_INDEX]);
         final int[] baseCountsCopy = Arrays.copyOf(baseCounts, baseCounts.length);
         baseCountsCopy[refBase.ordinal()] = -1;
-        final int altBaseIndex = MathUtils.argmax(baseCountsCopy);
+        final int altBaseIndex = MathUtils.maxElementIndex(baseCountsCopy);
         final boolean referenceSite = baseCounts[altBaseIndex] == 0;
 
-        /*** End heuristics ***/
-
         // if the site is ref, we simply update the coverage histogram
         if (referenceSite){
-            refSiteHistograms[contextToIndex(refContext)].increment(depth);
+            refSiteHistograms.get(refContext).increment(depth <= MAX_REF_DEPTH ? depth : MAX_REF_DEPTH);
             return;
         }
 
         // if we got here, we have an alt site
         final Nucleotide altBase = Nucleotide.valueOf(BaseUtils.baseIndexToSimpleBase(altBaseIndex));
 
-        final int[] altF1R2Counts = new int[4];
-        for (Nucleotide base : Arrays.asList(Nucleotide.A, Nucleotide.C, Nucleotide.G, Nucleotide.G)){
-            altF1R2Counts[base.ordinal()] = pileup.getNumberOfElements(pe ->
-                    Nucleotide.valueOf(pe.getBase()) == base && ! ReadUtils.isF2R1(pe.getRead()));
-        }
+        final int[] altF1R2Counts = regularBases.stream().mapToInt(base -> pileup.getNumberOfElements(
+                pe -> Nucleotide.valueOf(pe.getBase()) == base && ReadUtils.isF1R2(pe.getRead()))).toArray();
```

Comment:
Extract a method 
```
int f1r2BaseCounts(final Pileup pileup, final byte base) { 
  return pileup.getNumberOfElements( pe -> pe -> Nucleotide.valueOf(pe.getBase()) == base && ReadUtils.isF1R2(pe.getRead());
}
```

Reply:
Alternatively, you could do `altF1R2Counts = pileup.makeFilteredPileup(pe -> ReadUtils.isF1R2(pe.getRead()).baseCounts()` or something like that.